prusti-tests/tests/verify_overflow/pass/competitive_programming/greedy/FrogJumping.rs

```diff
@@ -0,0 +1,111 @@
+    }
+}
+
+// Recursive Solution
+
+#[pure]
+#[requires(positions.len() >= 2 && positions.len() <= 100000)]
+#[requires(idx >= 0 && idx < positions.len())]
+#[requires(next > idx && next <= positions.len())]
+#[requires(last_positions.len() == positions.len())]
+#[requires(idx < positions.len() - 1 ==> next <= last_positions.lookup(idx))]
+#[requires(positions.lookup(0) == 0)]
+#[requires(forall(|i: isize| (i >= 0 && i < positions.len() - 1) ==>  (last_positions.lookup(i) > i && last_positions.lookup(i) <= last_positions.lookup(i + 1) && last_positions.lookup(i) < positions.len())))]
+#[requires(last_positions.lookup(positions.len() - 1) == positions.len() - 1)]
+#[requires(forall(|i: isize, j: isize| (i >= 0 && i < positions.len() - 1 && j > i && j < positions.len()) ==>
+            (last_positions.lookup(i) <= last_positions.lookup(j))))]
+#[ensures(result < positions.len() - idx && result < 100000 && result >= 0)]
+#[ensures(result < positions.len() - idx && result < 100000 && result >= 0)]
+#[ensures (forall(|i: isize| (i > next && i <= last_positions.lookup(idx)) ==> 
+            result <= solve_rec(positions,  last_positions, idx, i)))]
+#[ensures((idx < positions.len() - 1 && next == last_positions.lookup(idx)) ==> 
+            result == solve_rec(positions, last_positions, next, next + 1) + 1)]
+#[ensures((idx < positions.len() - 1 && next == idx + 1) ==> 
+            result >= solve_rec(positions, last_positions, next, next + 1))]
+#[ensures(next == idx + 1 ==> forall(|j:  isize| (j > idx && j < positions.len()) ==> 
+            result >= solve_rec(positions, last_positions, j, j + 1)))]
+#[ensures(idx < positions.len() - 1 ==> result == solve_rec(positions, last_positions, idx, last_positions.lookup(idx)))]
+#[ensures(idx < positions.len() - 1 ==> 
+            result == solve_rec(positions, last_positions, last_positions.lookup(idx), last_positions.lookup(idx) + 1) + 1)]
+fn solve_rec(
+    positions: &VecWrapperI32,
+    last_positions: &VecWrapperI32,
+    idx: isize,
+    next: isize,
+) -> isize {
+    if idx == positions.len() - 1 {
+        0
+    } else if next == last_positions.lookup(idx) {
+        assert!(idx < positions.len() - 1 && next == last_positions.lookup(idx));
+        solve_rec(positions, last_positions, next, next + 1) + 1
+    } else {
+        assert!(idx < positions.len() - 1 && next < last_positions.lookup(idx));
+        let take = solve_rec(positions, last_positions, next, next + 1) + 1;
+        let leave = solve_rec(positions, last_positions, idx, next + 1);
+        let x1 = last_positions.lookup(idx) + 1;
+        let tmp1  = solve_rec(positions, last_positions, last_positions.lookup(idx), x1);
+        assert!(leave ==  tmp1 + 1);
+        let x2 = next + 1;
+        assert!(x2 ==  next + 1);
+        assert!(last_positions.lookup(idx) >  next && last_positions.lookup(idx) < positions.len());
+        let tmp2 = solve_rec(positions, last_positions, next, x2);
+        assert!(tmp2 >= tmp1);
```

Comment:
@vakaras This assertion is failing. But I think it shouldn't because the post condition at line 81 should be used to verify it. We have `x2 == next + 1`, that will lead to `forall(|j:  isize| (j > idx && j < positions.len()) ==> 
            result >= solve_rec(positions, last_positions, j, j + 1))` and by using `last_positions.lookup(idx)` as `j` then we should get `tmp2 >= tmp1`. Do you think I am missing something? Also, I tried to increase the assert timeout but it fails in less time.

Reply:
One thing that comes to my mind is that the quantifier on line 81 does not have a good trigger because `j+1` cannot be used inside a trigger. Did you try to use the trick with pulling out the addition into a separate pure function? You will probably also need to change `assert!(x2 ==  next + 1);` into `assert!(x2 ==  inc(next));`.

Reply:
I changed that and the verification succeeded.

Reply:
Great!